dashboard/lib/policies/ai.rb

```diff
@@ -0,0 +1,27 @@
+require 'user'
+
+class Policies::Ai
+  # Whether or not AI rubric features (AI TA) are enabled.
+  def self.ai_rubrics_enabled?(user)
+    return false if user.nil?
+
+    !user.ai_rubrics_disabled
+  end
+
+  # Whether or not AI features are enabled for a particular user and level.
+  def self.ai_rubrics_enabled_for_script_level?(user, script_level)
+    return false if user.nil?
+    return false if script_level.nil?
+
+    sections = user.sections_as_student
+    return false if sections.empty?
+
+    # Get sections that contain the given script_level (or aren't related to any unit)
+    sections = sections.select {|s| s.script.nil? || s.script.try(:id) == script_level.script.id}
+
+    # It is enabled if any of these sections have a teacher that has not opted-out
+    sections.any? do |section|
+      Policies::Ai.ai_rubrics_enabled?(section.user)
+    end
```

Comment:
Here, this enables AI rubric evaluation (and would then submit the job to the AI agent) if and only if the student:

* Is in any unit-less section where the teacher has not opted-out
* And/Or is in any related unit where that particular teacher has not opted-out

If the only teacher that has opted-out is leading a section that the student is in that has assigned a different unit than the one indicated by the level, it would still mark AI as enabled for this work.

If there is ANY teacher leading this student in a script-less section (no specifically assigned unit) that has not opted-out, the AI will evaluate that level.

(It might be nice to be able to know the section in the milestone and have section-specific AI settings)